src/mesh/mesh-adapter.test.ts

```diff
@@ -0,0 +1,60 @@
+import { MeshTag, MeshTenant } from "./mesh-tenant.model.ts";
+import { MeshAdapter } from "./mesh-adapter.ts";
+import { assertEquals } from "../dev-deps.ts";
+
+Deno.test("mesh-adapter can see which tags have changed compared to the original", () => {
```

Comment:
d: I did not comment on the other files but this is my general suggestion: don't make this function part of the interface (or better keep it as an interface). Technically you are not supposed to unit test internal methods. This makes refactoring harder and classes should be treated as black boxes. Give them some input and check what state transition they had or what other APIs they call and check if this matches the expectations. This test class now is an attempt to surface an otherwise hidden method and you did this as a shortcut to avoid testing all three big implementations of the facades. It also is a somewhat unrelated functionality for this interface as it is just related to the tagging stuff. So you also see this is a not-so-good design decision as it now adds functionality/dependency from a tag "namespace" to the general "mesh" namespace.
A cleaner approach IMO would be to have a own unit/class which does this consolidation and inject it into the affected MeshAdapter implementations. You could then cleanly unit test this class via its public method `getChangedTags()` and on the other MeshAdapter implementations unit test you could provide a mock and assert that this method was called during the invocation of the `updateTenant` call (because this would be now actually impossible to test and you would need to check the correct working of this call for all 3 implementations when done properly :D).
One can argue that by injecting a new class into the implementations it also introduces a cross dependency towards the `tag` namespace of the application to the adapters, which is true, but I think its easier to revert as its not so deeply "woven" into the code and only an injected class and thus easier to replace or revert.